docs/DeveloperGuide.md

```diff
@@ -1,29 +1,204 @@
+
+
+### Manager Component (PeiHao)
+The manager component consists of four different managers, 
+in which the list of entity is initialized and the methods implementing the entity are written inside.
+- DeadlineManager: This class contains an ArrayList of deadlines and methods implementing the deadlines like addDeadline, printDeadline and deleteDeadline.
+- DishManager: This class contains an ArrayList of dishes and methods like addDish, deleteDish, viewDish and 
+stringOfDish which returns the dish information.
+- MeetingManager: This class contains an ArrayList of meetings and methods like addMeeting, printMeetings and deleteMeeting.
+- StaffManager: This class contains an ArrayList of staffs and methods like addStaff, deleteStaffByName, findStaffByName and getStaffString which print all the staffs' information.
+
+### Ui Component (HuiQi)
+The Ui Component consists of the TextUi class that handles interactions between the app and the user.
+
+The `TextUi`class performs the following functions:
+1. Reading the user input (Empty inputs will be ignored)
+
+    method: `getUserInput()`
+2. Printing messages generated by the app to the terminal
+
+    method: `printBanner()`, `printMessage()`
+
+### Utils Component (Darren)
+
+![UtilsParserClassDiagram](uml/images/UtilsParserClassDiagram.png)    
+The Utils Component consists of the `Parser` class that will handle the parsing and preparing of commands within the DinerDirector application.  
+
+The `Parser` class performs the following functions:  
+* Parse the command given the user input and extracts out the necessary information related to the command.
+* Returns the appropriate Command Class based on the parsed input.
+
+![UtilsStorageClassDiagram](uml/images/UtilsStorageClassDiagram.png)  
+The Utils Component also consists of the `Storage` class and the individual `XYZStorage` that will handle storage related operations within the DinerDirector application.  
+
+The `Storage` class performs the following functions:
+* Create a directory in the root folder of where the application is running.
+
+The `XYZStorage` class performs the following functions:
+* Read and load data from the file.
+* Write to the file if any changes occur to the list.
+
+## Implementation
+### Parsing Feature (Darren)
+
+![ParserSquenceDiagram](./uml/images/ParserSequenceDiagram.png)  
+How the parsing works:
+1. The `Parser()` class will be called to create a new instance of `Parser`.
+2. Afterwards, when the `parseCommand()` method is called from `DinerDirector` class, the `parseCommand()` will split the given userInput first.
+2. With the `userInputSplit[]`, the `0` index will be extracted out. That will be used as identification for the command the user typed in.
+3. The `commandWord` will be used in the switch statement to select the appropriate command. returning `IncorrectCommand` class is the default behavior.
```

Comment:
there seems to be a link to the Command class missing from the sequence diagram.